src/hotspot/share/opto/macro.cpp

```diff
@@ -626,9 +626,11 @@ bool PhaseMacroExpand::can_eliminate_allocation(AllocateNode *alloc, GrowableArr
           DEBUG_ONLY(disq_node = use;)
           NOT_PRODUCT(fail_eliminate = "null or TOP memory";)
           can_eliminate = false;
-        } else {
-          safepoints.append_if_missing(sfpt);
+        } else if (safepoints != nullptr) {
+          safepoints->append_if_missing(sfpt);
         }
+      } else if (ignore_merges && (use->is_Phi() || use->is_EncodeP() || use->Opcode() == Op_MemBarRelease)) {
```

Comment:
I try to understand this part. now `can_eliminate_allocation` can pre-test whether SR can eliminate the alloc.  I see that you use it in EA. 

With ignore_merges, why we also skip EncodeP or MemBarRelease here?

Reply:
Do you really need the boolean parameter ignore_merges here? 
It looks like we can use (safepoints == nullptr) instead?